status/controller.go

```diff
@@ -23,24 +25,52 @@ import (
 )
 
 type Controller[T Object] struct {
-	kubeClient         client.Client
-	eventRecorder      record.EventRecorder
-	observedConditions sync.Map // map[reconcile.Request]ConditionSet
-	terminatingObjects sync.Map // map[reconcile.Request]DeletionTimestamp
+	objectGVK                     schema.GroupVersionKind
+	kubeClient                    client.Client
+	eventRecorder                 record.EventRecorder
+	observedConditions            sync.Map // map[reconcile.Request]ConditionSet
+	terminatingObjects            sync.Map // map[reconcile.Request]DeletionTimestamp
+	ConditionDuration             pmetrics.ObservationMetric
+	ConditionCount                pmetrics.GaugeMetric
+	ConditionCurrentStatusSeconds pmetrics.GaugeMetric
+	ConditionTransitionsTotal     pmetrics.CounterMetric
+	TerminationCurrentTimeSeconds pmetrics.GaugeMetric
+	TerminationDuration           pmetrics.ObservationMetric
 }
 
-func NewController[T Object](client client.Client, eventRecorder record.EventRecorder) *Controller[T] {
+type ControllerOpts struct {
```

Comment:
We need to be able to pass in the type for client.object, since unstructured types do not have a Kind.